### PR TITLE
Tracking missing `route_id` in payments

### PR DESCRIPTION
**Pull request by charlie-costanzo**
# Description

Recently, agencies have been observing a large amount of the `Route Z` label for `route_id` in payments data, which is Littlepay's default when they can't associate a route with a payment.

This PR seeks to create the necessary underlying tables to introduce preliminary testing and alerts to track high prevalence of `Route Z` and null values in our Metabase dashboard.

This PR also adds the `day_of_week` calculation to `payments_rides` as requested by SBMTD in #2342 

**Note to discuss**
- [x] This PR removes the automatic coalescing to `Route Z` for null values in `payments_rides` in favor of preserving them to see the proportion of originally `Null` values alongside `Route Z` values in the source data

To-do: 
- [x] yml documentation for new table

Resolves # [issue]

## Type of change

- [x] New feature

## How has this been tested?
locally with `dbt run` and `dbt test`

### CODE REVIEW

File: warehouse/models/payments_views/payments_rides.sql [hunk 1]
```diff
@@ -294,11 +294,7 @@ join_table AS (
         t2.latitude AS off_latitude,
         t2.longitude AS off_longitude,
         t2.geography AS off_geography,
-        CASE
-            WHEN
-                t1.route_id != 'Route Z' THEN t1.route_id
-            ELSE COALESCE(t2.route_id, 'Route Z')
-        END AS route_id
+        COALESCE(t1.route_id, t2.route_id, Null) AS route_id
```
Comment (atvaccaro) [inline warehouse/models/payments_views/payments_rides.sql:297]: nit: you don't need a final `null`, that's already what coalesce will end up with if all inputs are null

File: warehouse/models/payments_views/payments_rides.sql [hunk 2]
```diff
@@ -326,13 +322,7 @@ join_table AS (
...
             -- (don't have to handle unkowns the way we do with route_id)
             AND EXTRACT(DATE FROM TIMESTAMP(t1.transaction_date_time_utc)) = r.date
-            AND r.route_id = (
-                CASE
-                    WHEN
-                        t1.route_id != 'Route Z' THEN t1.route_id
-                    ELSE COALESCE(t2.route_id, 'Route Z')
-                END
-            )
+            AND r.route_id = COALESCE(t1.route_id, t2.route_id, Null)
```
Comment (atvaccaro) [inline warehouse/models/payments_views/payments_rides.sql:325]: same here

File: warehouse/models/payments_views/tests/payments_reliability_mislabeled_routes.sql [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+count_route_z_rides AS (
+    SELECT
+
+        participant_id,
+        month_start,
+        COUNT(*) AS n_route_z_rides
+
+    FROM payments_tests_monthly_date_spine
+    INNER JOIN route_z_rides
+        USING (participant_id) WHERE transaction_date_pacific >= month_start AND transaction_date_pacific <= month_end
```
Comment (atvaccaro) [inline warehouse/models/payments_views/tests/payments_reliability_mislabeled_routes.sql:28]: is there a reason to use `using` and `where` over `ON ... AND ...`? same elsewhere

File: warehouse/models/payments_views/tests/payments_reliability_mislabeled_routes.sql [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        COALESCE(null_rides.n_null_rides, 0) AS n_null_rides,
+        (COALESCE(z_rides.n_route_z_rides, 0) + COALESCE(null_rides.n_null_rides, 0)) AS total_mislabeled_rides,
+        COALESCE(all_rides.n_all_rides, 0) AS n_all_rides
+
+    FROM payments_tests_monthly_date_spine AS date_spine
+    LEFT JOIN count_all_rides AS all_rides
+        USING (participant_id, month_start)
+    LEFT JOIN count_route_z_rides AS z_rides
+        USING (participant_id, month_start)
+    LEFT JOIN count_null_rides AS null_rides
```
Comment (atvaccaro) [inline warehouse/models/payments_views/tests/payments_reliability_mislabeled_routes.sql:73]: You should be able to do these all at once in the aggregation; I'd recommend doing it for practice. You can use conditions to count different columns; for example [COUNTIF](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#countif) to check for specific values. `count(column)` already only counts not-null rows for that column; `count(0)/count(*)` counts the row regardless.

File: warehouse/models/payments_views/tests/payments_reliability_mislabeled_routes.sql [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        USING (participant_id, month_start)
+),
+
+calculate_relative_count AS (
+    SELECT
+
+        *,
+
+        SAFE_DIVIDE(total_mislabeled_rides, n_all_rides) * 100 AS relative_mislabeled_rides_to_total_rides,
```
Comment (atvaccaro) [inline warehouse/models/payments_views/tests/payments_reliability_mislabeled_routes.sql:82]: nit: can the name indicate it's a percentage?

File: warehouse/models/payments_views/tests/_payments_views_tests.yml [hunk 6]
```diff
@@ -31,3 +31,21 @@ models:
...
+        description: Littlepay-assigned Participant ID
+      - name: month_start
+        description: The first day of the month for which the count aggregation is being caclulated
+      - name: n_route_z_rides
+        description: The total number of rides labeled as `Route Z` for the given month
+      - name: n_null_rides
+        description: The total number of rides labeled as `Null` for the given month
+      - name: total_mislabeled_rides
+        description: The total number of rides labeled as either `Route Z` or `Null` for the given month
+      - name: n_all_rides
```
Comment (atvaccaro) [inline warehouse/models/payments_views/tests/_payments_views_tests.yml:46]: I'd add a test that `n_all_rides` is greater than the sum of the other two

File: warehouse/models/payments_views/tests/payments_reliability_mislabeled_routes.sql [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+            month_start,
+            n_route_z_rides,
+            n_null_rides,
+            total_mislabeled_rides,
+            n_all_rides,
+            relative_mislabeled_rides_to_total_rides,
+
+            RANK() OVER (PARTITION BY participant_id ORDER BY month_start DESC) AS recency_rank
+
+        FROM calculate_relative_count)
```
Comment (atvaccaro) [inline warehouse/models/payments_views/tests/payments_reliability_mislabeled_routes.sql:111]: I don't think you need this subquery if you're adding a window function column

**Review by atvaccaro — CHANGES_REQUESTED** [review]

File: warehouse/models/payments_views/tests/_payments_views_tests.yml (hunk 6 above)
Comment (charlie-costanzo) [reply warehouse/models/payments_views/tests/_payments_views_tests.yml:46]: @atvaccaro right now we have some participants where `n_all_rides = (n_route_z_rides + n_null_rides)` and I don't know when/if that will be resolved. Would it make sense to test that `n_all_rides` is greater than *or equal to* the other two columns, or would that defeat the purpose?

**Comment by charlie-costanzo** [comment]
@atvaccaro thank you for the review! I just made the suggested revisions for whenever you're able to take a look

File: warehouse/models/payments_views/tests/_payments_views_tests.yml (hunk 6 above)
Comment (atvaccaro) [reply warehouse/models/payments_views/tests/_payments_views_tests.yml:49]: Oh sure sorry, `>=` would make sense. I just mean we aren't overcounting in some weird way

**Review by atvaccaro — APPROVED** [review]
lgtm!

**Pull request merged**